Algorithm.Python/EmaCrossFuturesFrontMonthAlgorithm.py

```diff
@@ -0,0 +1,104 @@
+### <meta name="tag" content="using data" />
+### <meta name="tag" content="indicator" />
+### <meta name="tag" content="futures" />
+class EmaCrossFuturesFrontMonthAlgorithm(QCAlgorithm):
+
+    def Initialize(self):
+        self.SetStartDate(2013, 10, 8)
+        self.SetEndDate(2013, 10, 10)
+        self.SetCash(1000000)
+
+        future = self.AddFuture(Futures.Metals.Gold);
+
+        # Only consider the front month contract
+        future.SetFilter(lambda x: x.FrontMonth().OnlyApplyFilterAtMarketOpen())
+
+        # Symbol of the current contract
+        self.symbol = None
+
+        # Create two exponential moving averages
+        self.fast = ExponentialMovingAverage(100)
+        self.slow = ExponentialMovingAverage(300)
+        self.tolerance = 0.001
+        self.consolidator = None
+
+        # Add a custom chart to track the EMA cross
+        chart = Chart('EMA Cross')
+        chart.AddSeries(Series('Fast', SeriesType.Line, 1))
+        chart.AddSeries(Series('Slow', SeriesType.Line, 1))
+        self.AddChart(chart)
+
+    def OnData(self,slice):
+
+        holding = None if self.symbol is None else self.Portfolio.get(self.symbol)
+        if holding is not None:
+            # Buy the futures' front contract when the fast EMA is above the slow one
+            if self.fast.Current.Value > self.slow.Current.Value * (1 + self.tolerance):
+                if not holding.Invested:
+                    self.SetHoldings(self.symbol, .1)
+                    self.PlotEma()
+            elif holding.Invested:
+                self.Liquidate(self.symbol)
+                self.PlotEma()
+
+    def OnSecuritiesChanged(self, changes):
+        if len(changes.RemovedSecurities) > 0:
+            # Remove the consolidator for the previous contract
+            # and reset the indicators
+            if self.symbol is not None and self.consolidator is not None:
+                self.SubscriptionManager.RemoveConsolidator(self.symbol, self.consolidator)
+                self.fast.Reset()
+                self.slow.Reset()
+            # We don't need to call Liquidate(_symbol),
+            # since its positions are liquidated because the contract has expired.
+
+        # Only one security will be added: the new front contract
+        self.symbol = changes.AddedSecurities[0].Symbol
+
+        # Create a new consolidator and register the indicators to it
+        self.consolidator = self.ResolveConsolidator(self.symbol, Resolution.Minute)
+        self.RegisterIndicator(self.symbol, self.fast, self.consolidator)
+        self.RegisterIndicator(self.symbol, self.slow, self.consolidator)
+
+        # Update the consolidator with historical data
+        # This will update the registered indicators
+        history = self.History(self.symbol, self.slow.WarmUpPeriod)
+        for index, row in history.iterrows():
+            time = index[2]
+            bar = TradeBar(time, self.symbol, row.open, row.high, row.low, row.close, 0)
```

Comment:
We could also use the WarmUpIndicator() helper method API?

Reply:
Great idea. It's particular useful in Python to avoid pandas conversion.